### PR TITLE
Support for Jasmine sessions for BrowserStack Test Observability (v7)

### PR DESCRIPTION
**Pull request by sriteja777**
## Proposed changes

Add support for Jasmine sessions to be captured by BrowserStack Test Observability

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browserstack-service/src/util.ts [hunk 1]
```diff
@@ -318,7 +318,11 @@ export async function getGitMetaData () {
     }
 }
 
-export function getUniqueIdentifier(test: Frameworks.Test): string {
+export function getUniqueIdentifier(test: Frameworks.Test, framework?: string): string {
+    console.log('came getIsdsdsdgs')
```
Comment (amaanbs) [inline packages/wdio-browserstack-service/src/util.ts:322]: Was this added for debugging purposes? If yes, please remove.

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 2]
```diff
@@ -46,34 +64,44 @@ export default class InsightsHandler {
...
-            if (this._tests[fullTitle]) {
-                this._tests[fullTitle].finishedAt = (new Date()).toISOString()
-            } else {
-                this._tests[fullTitle] = {
-                    finishedAt: (new Date()).toISOString()
-                }
+        if (!frameworkSupportsHook('after', this._framework)) {
+            return
+        }
+        console.log('came heree')
```
Comment (amaanbs) [inline packages/wdio-browserstack-service/src/insights-handler.ts:86]: Remove debug logs

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 3]
```diff
@@ -46,34 +64,44 @@ export default class InsightsHandler {
...
+
+        const fullTitle = getUniqueIdentifier(test, this._framework)
+        console.log('fulltitle is ', fullTitle)
```
Comment (amaanbs) [inline packages/wdio-browserstack-service/src/insights-handler.ts:89]: Remove debug logs

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 4]
```diff
@@ -46,34 +64,44 @@ export default class InsightsHandler {
...
+        if (this._tests[fullTitle]) {
+            this._tests[fullTitle].finishedAt = (new Date()).toISOString()
+        } else {
+            this._tests[fullTitle] = {
+                finishedAt: (new Date()).toISOString()
             }
-            await this.sendTestRunEvent(test, 'HookRunFinished', result)
         }
+
+        console.log('this._tests is ', this._tests)
```
Comment (amaanbs) [inline packages/wdio-browserstack-service/src/insights-handler.ts:99]: Remove debug logs

**Review by amaanbs — CHANGES_REQUESTED** [review]

**Comment by sriteja777** [comment]
@amaanbs Sorry, have forgot to remove them. Removed them now. Please re-review

**Pull request closed without merging**